stock_request/models/stock_request_order.py

```diff
@@ -317,6 +317,16 @@ def _check_location_company(self):
                 )
             )
 
+    @api.constrains("stock_request_ids")
```

Comment:
IMHO, this is too restricting. As you maybe want to begin a stock request before filling it in. I'm not in favor of this

Reply:
There's no sense in having an empty one, Denis. If you want to do it later, just do it later. Stock request orders are literally a grouping over stock requests.

Reply:
Yes, but this seems a little bit restrictive. Nothing in Odoo restricts void purchases, sales, pickings...

Reply:
They are not exactly the same type of container. This solution comes after the discussion in #1949, as the less bad option. Please check that one.

Reply:
Yes, I didn't follow the end conversation and you merged fast.

So, this refrains people to void a draft request order and then reflll it in after ?

Reply:
@victoralmau can you comment the steps to reproduce the problem? Basically it's to remove the orders in an existing one.

Reply:
Maybe an explanation will help to understand it better.

![ejemplo](https://github.com/OCA/stock-logistics-warehouse/assets/4117568/649b6fd0-b2d5-4853-aa4b-2671c7d8b57f)

If we have a stock request order already created with lines, we can delete some of them, but not all. If we want to delete all of them, the process will be to cancel the stock request order.

Reply:
Ughhh, yes but this is the change you introduce. Not the current behaviour. :sweat_smile:

Reply:
The current behavior is as indicated in the previous PR, deleting all the lines of a stock request order would cause it to be set in "_draft_" status but could not be cancelled, causing confusion.

Reply:
In the other PR, the possibility of changing the compute was raised and discarded. The status of the stock request order is and must be that of the lines that contain it, and to avoid not being able to cancel an empty stock request order, precisely this change is added.